Source/WebKit/UIProcess/Cocoa/WKStorageAccessAlert.mm

```diff
@@ -41,6 +41,67 @@
 #import <WebCore/RegistrableDomain.h>
 #import <wtf/BlockPtr.h>
 
+#if PLATFORM(MAC)
+@interface _WKSSOSiteList : NSObject<NSTableViewDataSource, NSTableViewDelegate>
+- (instancetype)initWithSiteList:(NSArray<NSString *> *)siteList withAlert:(RetainPtr<NSAlert>)alert withTableView:(RetainPtr<NSTableView>)tableView;
+- (IBAction)toggleTableViewContents:(NSButton *)sender;
+@end
+
+@implementation _WKSSOSiteList {
+    RetainPtr<NSArray<NSString *>> _siteList;
+    RetainPtr<NSAlert> _alert;
+    RetainPtr<NSTableView> _tableView;
+}
+
+- (instancetype)initWithSiteList:(NSArray<NSString *> *)siteList withAlert:(RetainPtr<NSAlert>)alert withTableView:(RetainPtr<NSTableView>)tableView
+{
+    if (!(self = [super init]))
+        return self;
+
+    _siteList = adoptNS([[NSArray alloc] initWithArray:siteList copyItems:YES]);
+    _alert = alert;
+    _tableView = tableView;
+    return self;
+}
+
+- (NSInteger)numberOfRowsInTableView:(NSTableView *)aTableView
+{
+    return _siteList.get().count;
+}
+
+- (NSView *)tableView:(NSTableView *)tableView viewForTableColumn:(NSTableColumn *)tableColumn row:(NSInteger)row
+{
+    NSString *viewIdenfier = [NSString stringWithFormat:@"row%ldIdentifier", (long)row];
+    RetainPtr result = [tableView makeViewWithIdentifier:viewIdenfier owner:self];
+    if (result == nil) {
```

Comment:
Nit - we generally prefer `!result` instead of checking explicitly against `nil`